### ISSUE TITLE
aws_eip should have a Public DNS attribute

### ISSUE THREAD

**Issue by colebrumley** [terraform-providers/terraform-provider-aws#1149]
Elastic IPs are assigned a Public DNS name. That name should be accessible in Terraform as an attribute of `aws_eip` objects. The final bullet of [this](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#eip-basics) Elastic IP doc reads:

> We resolve a public DNS hostname to the public IPv4 address or the Elastic IP address of the instance outside the network of the instance, and to the private IPv4 address of the instance from within the network of the instance.

Using the [DNS name](https://forums.aws.amazon.com/thread.jspa?messageID=99701#99701) is the recommended method to use an eip to reference internal resources.

### Terraform Version
Terraform v0.9.11

### Affected Resource(s)
- aws_eip

### Important Factoids
Although the `describe-addresses` API call does not specifically contain a field with the public DNS, the AWS console does, and the IP does not need to be associated to anything first. I'm not sure if there is another API call that would tell you the DNS name or if the console is just templating out the `ip-0-0-0-0.compute-1.amazonaws.com` format.

Editing to add an additional AWS forum link that's dated but specifies that it is a derivation of the IP and is not available from an API: https://forums.aws.amazon.com/message.jspa?messageID=84856

**Comment by phistrom** [comment]
`ip-0-0-0-0.`**`compute-1`**`.amazonaws.com` is for Elastic IP public DNS in **`us-east-1`**. In other regions like **`us-west-2`**, it appears to be `ip-0-0-0-0.`**`us-west-2.compute`**`.amazonaws.com`



The following is a workaround until this seemingly simple attribute gets added to the aws_eip resource.
```
data "aws_region" "current" {}
locals {
  elastic-ip-domain = "${data.aws_region.current.name == "us-east-1" ? "compute-1" : format("%s.compute", data.aws_region.current.name)}"
}
resource "aws_route53_record" "some-cname" {
  name = "some-host"
  type = "CNAME"
  ttl = 300
  records = ["ec2-${replace(aws_eip.some-eip.public_ip, ".", "-")}.${local.elastic-ip-domain}.amazonaws.com"]
  # records = ["${aws_instance.instance.public_dns}"]  # this requires 2 invocations to get right because of Elastic IP
  zone_id = "${var.some-route53-zone}"
}
resource "aws_eip" "some-eip" {
  vpc = true
}
```
The CNAME is preferred because an internal resource will resolve it to the private IP address instead of the public one.

**Comment by bflad** [comment]
Support for the `public_ip` and `private_ip` (when available) attributes has been merged into the `aws_eip` data source and resource. This will release with version 2.2.0 of the Terraform AWS Provider shortly.

**Comment by bflad** [comment]
This has been released in [version 2.2.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#220-march-15-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by peterromfeldhk** [comment]
still getting `aws_eip does not have attribute public_dns`

```
$ terraform version
Terraform v0.11.11
+ provider.archive v1.2.1
+ provider.aws v1.43.2
```
